tox.ini

```diff
@@ -184,3 +184,9 @@ commands =
                 --hidden-import pytest_doctestplus.plugin \
                 --hidden-import pytest_mpl.plugin
     ./run_astropy_tests --astropy-root {toxinidir}
+
+[testenv:mypy]
+deps =
+    mypy
+commands =
+    mypy {tox_root}/astropy --install-types --non-interactive
```

Comment:
There are some packages that aren't typed themselves, but have created type stub packages that contain type information.  The [`--install-types`](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-install-types) flag "causes mypy to install known missing stub packages for third-party libraries using pip", which means that mypy would have access to verified type information for packages like pandas, PyYAML, beautifulsoup4, etc.  We could also manually add other type stub packages here, or otherwise specify mypy not to follow certain imports.